readthedocs/projects/notifications.py

```diff
@@ -11,3 +15,69 @@ class ResourceUsageNotification(Notification):
     context_object_name = 'project'
     subject = 'Builds for {{ project.name }} are using too many resources'
     level = REQUIREMENT
+
+
+class DeprecatedWebhookEndpointNotification(Notification):
```

Comment:
This notification could be generalized in the future so other notifications can extend from the base class and use the same workflow for de-duplication and sending emails.